### ISSUE TITLE
Nested nullability is dropped from constraint when type is from different assembly

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#29048]
No warning reported for `new A<string>()` though `string` does not satisfy the `IEquatable<string?>` constraint.

`csc /langversion:8.0 /t:library a.cs`
```c#
using System;
public class A<T> where T : IEquatable<string?>
{
}
```
`csc /langversion:8.0 /r:a.dll b.cs`
```c#
class B
{
    static void Main()
    {
        object o = new A<string>(); // warning
    }
}
```
